### PR TITLE
execute already registered command in the same tick

### LINKED ISSUE #7542: Cut/Copy/Paste Keyboard Shortcuts Work Intermittently
### Description
 
Once in a while, particularly after loading Theia (in Gitpod for me), the keyboard shortcuts for Cut/Copy/Paste don't work. For instance, when cutting text, the text just stay highlighted.

### Reproduction Steps

1. Pick a file that can be opened with Theia's text editor
2. Select text
3. Press CTRL-X, CTRL-C, or CTRL-V

**OS and Theia version:**
Firefox 73.0 on macOS Catalina and Firefox 75.0 on Ubuntu 19.10
Theia from Gitpod (I don't know how to get the Theia version from Gitpod)

**Diagnostics:** 

I didn't see errors in the browser console.
This issue seems to be rectified after selecting the desired control using the right-click context menu, then later trying the keyboard shortcut again.

cc @jankeromnes

### PR DESCRIPTION
**Pull request by akosyakov**
#### What it does

- fix #7542: execute already registered command in the same tick, otherwise FF cannot detect that it is triggered by a user action and denies calls to document.executeCommand

#### How to test

- use FF browser, test that you can copy/paste/cut in different input fields and editor

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
Not so often, but it still happens :(

This time is for sure the same tick:
Screenshot 2020-05-20 at 17 02 45

But FF still complaints about short-lived user triggered event. Is it stack trace length or timing based? 🤷

**Comment by akosyakov** [comment]
@eclipse-theia/ecd-theia-committers If someone knows what FF means under `short-lived user triggered event handler` please help? It apparently does not mean code of event handler running in the same tick. Or maybe I'm overlooking something in the implementation?

File: packages/core/src/common/command.ts [hunk 1]
```diff
@@ -285,24 +286,34 @@ export class CommandRegistry implements CommandService {
...
      */
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    async executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
+    executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
         const handler = this.getActiveHandler(commandId, ...args);
-        if (handler) {
-            await this.fireWillExecuteCommand(commandId, args);
-            const result = await handler.execute(...args);
+        if (!handler) {
+            const argsMessage = args && args.length > 0 ? ` (args: ${JSON.stringify(args)})` : '';
```
Comment (akosyakov) [inline packages/core/src/common/command.ts:296]: (off-topic) but serializing of random args won't work, just imagine widget 

we have to remove this code

**Comment by lmcbout** [comment]
Start testing with FF andd the first command "CUT" in a readme.md file generated the following warning:
document.execCommand('cut'/'copy') was denied because it was not called from inside a short running user-generated event handler. common-frontend-contribution.ts:502:29

File: packages/core/src/common/command.ts (hunk 1 above)
Comment (vzhukovskii) [reply packages/core/src/common/command.ts:296]: as I understand, the args count might be infinite and one of this arg cam represent a widget, am I right?

**Comment by akosyakov** [comment]
Wonder why it was working before, maybe a regression in newer ff version?

**Comment by marechal-p** [comment]
> Not so often, but it still happens :(

Maybe because of the following async function? https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/command.ts#L290-L301

If the code goes through any `await` statement, the execution will most likely run on a different tick...

**Comment by spoenemann** [comment]
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard

In case nothing else works we could request the user's permission?

**Comment by marcdumais-work** [comment]
> https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard
> 
> In case nothing else works we could request the user's permission?

I saw that too, but looks like it applies when developing Firefox add-ons, where the user can be prompted to accept giving this permission before installing?

**Comment by spoenemann** [comment]
Oh, right.

**Comment by akosyakov** [comment]
So it is timeout based: https://github.com/mozilla/gecko-dev/blob/fc91a093e40dde71d10ad219946b8ae775aca9eb/docshell/base/BrowsingContext.cpp#L1058-L1076

Although for me the default timeout `dom.user_activation.transient.timeout` is 5s:
Screenshot 2020-05-25 at 11 03 48

That should be more than enough.

**Comment by akosyakov** [comment]
Timeous was added to FireFox only in Jan this year: https://github.com/mozilla/gecko-dev/commit/ff965689f51f9c26a3a25ec248034c41ff1eef19 that's why it was working before

**Comment by akosyakov** [comment]
I think we should file an issue for FF to clarify.

But generally keeping keydown processing fast and sync to make sure that typing is responsive, especially since we handle all typing to Monaco editor.

**Comment by akosyakov** [comment]
Relevant issue in FF: https://bugzilla.mozilla.org/show_bug.cgi?id=1634273

**Pull request closed without merging**